Source/Protocols/Client.swift

```diff
@@ -4,4 +4,6 @@ import Result
 
 public protocol Client {
   func performRequest<T: Request>(request: T, completionHandler: Result<T.ResponseObject, T.ResponseError> -> ()) -> NSURLSessionDataTask
+  func clearResponse(httpResponse: HTTPResponse) -> Result<NSData?, SwishError>
```

Comment:
Yeah, I wasn't totally sure. This PR kind of has a theme of "extreme flexibility with sensible defaults" and that's why I chose to add it to the protocol; the default APIClient just passes along 2xx, but other implementers can choose to do whatever they wish... thoughts?